analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs

```diff
@@ -49,6 +49,7 @@ public abstract class UriShouldNotBeHardcodedBase<TSyntaxKind, TLiteralExpressio
                 "FILE",
                 "PATH",
                 "URI",
+                "URIS",
```

Comment:
What about the pluralization of the other `CheckedVariableNames`? Couldn't we do a type lookup and if the type implements `IEnumerable` we allow an `s` at the end (files, paths, uris, and so on) of the variable name. This should also work for the `var uris = new List<string> { "C:/test.txt" }` cases.

Reply:
Is it really necessary?
I would not consider `var uris = "www.myuri.com";` a FP, so I'm not sure I see the value of additional checks that might or might not slow down the rule. What about `var uri = new List<string> { "C:/test.txt" }`, should we consider it compliant?

Reply:
URLS, URNS, and STREAMS are missing

Reply:
Yes, but would you consider `var uris = "www.myuri.com";` a FP?

Reply:
Yes. But I also wouldn't care too much about it. You can fix the issue like you did by simply adding "URIS".